indico/modules/rb_new/controllers/backend/bookings.py

```diff
@@ -258,13 +276,33 @@ def _process(self):
         return jsonify(booking_id=booking_id, room_id=room_id)
 
 
-class RHBookingEventData(RHBookingBase):
+class RHLinkedObjectData(RHRoomBookingBase):
+    """Fetch data from event, contribution or session block"""
+    def _process_args(self):
+        type_ = request.view_args['type']
+        id_ = request.view_args['id']
+        obj = {
+            'event': lambda x: Event.get_one(x),
+            'contribution': lambda x: Contribution.get_one(x),
+            'session_block': lambda x: SessionBlock.get_one(x)
+        }[type_](id_)
+
+        if type_ is 'event':
+            event = obj
+        else:
+            event = obj.event
+
+        if event.is_deleted:
+            raise ValidationError(_('The associated event is deleted'))
```

Comment:
`ValidationError` seems wrong here, we aren't in any validation function. what about just handling this like an object with no access?